### DISCUSSION TITLE
Add an  option to override a bundle package with local version

### DISCUSSION THREAD

**Discussion by llde** [tectonic-typesetting/tectonic#922] [Ideas]
Hi.
I was wondering if it's possible to add a mechanism to override a package shipped in the bundle with a local copy. This would help with developing the package. 
The use case would be for maintainers of packages that are in the bundle.  Currently they can use the -Z search-path to point to a local package or document class, however these, or at least the document classes, must have  a different name. With an ovverride option the local file would override the one in the bundle without renaming, simplifying the process. 

This would simplify testing and maintaining. 
The option could work also in an isolated or global ways: it could specify if it override only the package from the "init" document (the one you start compilation), the first and other local packages only, or also overriding it for all package in the bundle.
I found myself in the process of modifying some packages in local, it could have been shelpful.

Thanks for reading.

**Comment by pkgw** [comment]
If you use the `-Z search-path` unstable option, or just include the updated files next to your document source, you should be able to override bundle files without needing to rename them. Are you encountering a case where that doesn't seem to be happening?

**Reply by llde** [reply]
yes -Z search path seems to only find cls files with names different then the ones in the bundle. If they aqre equals the bundled one  win. I did try with latest git at the beginning of august, not sure if stuffs are changed.

**Reply by pkgw** [reply]
I've tried a few tests with the current version of Tectonic but everything seems to be working as expected in what I'm trying. Are you able to show an example of the setup that you're using where it looks like a file in the search path isn't overriding a bundle file?
